pkg/protocol/engine/ledgerstate/slot_diff.go

```diff
@@ -0,0 +1,178 @@
+package ledgerstate
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"sort"
+
+	"github.com/iotaledger/hive.go/kvstore"
+	"github.com/iotaledger/hive.go/serializer/v2/marshalutil"
+	iotago "github.com/iotaledger/iota.go/v4"
+)
+
+// SlotDiff represents the generated and spent outputs by a slot's confirmation.
+type SlotDiff struct {
+	// The index of the milestone.
+	Index iotago.SlotIndex
+	// The outputs newly generated with this diff.
+	Outputs Outputs
+	// The outputs spent with this diff.
+	Spents Spents
+}
+
+func slotDiffKeyForIndex(index iotago.SlotIndex) []byte {
+	m := marshalutil.New(9)
+	m.WriteByte(StoreKeyPrefixSlotDiffs)
+	m.WriteBytes(index.Bytes())
+
+	return m.Bytes()
+}
+
+func (ms *SlotDiff) KVStorableKey() []byte {
+	return slotDiffKeyForIndex(ms.Index)
+}
+
+func (ms *SlotDiff) KVStorableValue() []byte {
+	m := marshalutil.New(9)
```

Comment:
This serialization stuff we might be able to simplify a bit using serix and maybe a typed KV store. But no need to do it now imo.